library/src/blas1/rocblas_dot.hpp

```diff
@@ -29,12 +29,8 @@ __global__ void dot_kernel_part1(rocblas_int    n,
     // bound
     if(tid < n)
     {
-        x = load_ptr_batch(xa, hipBlockIdx_y, offsetx, stridex);
-        y = load_ptr_batch(ya, hipBlockIdx_y, offsety, stridey);
-        if(incx < 0)
-            x -= ptrdiff_t(incx) * (n - 1);
-        if(incy < 0)
-            y -= ptrdiff_t(incy) * (n - 1);
+        x       = load_ptr_batch(xa, hipBlockIdx_y, shiftx, stridex);
```

Comment:
I don't see an update for load_ptr_batch.   It should take a ptr_diff_t for the shift argument. 
When inlining the compiler may figure it out but better to be safe.